### PR TITLE
Fct jl

### PR DESCRIPTION
**Pull request by jakobliebmann**
hi Carl,
hi all,

restored the file forcats.Rmd, restored the intro, proposed a chapter-structure and did some work about general things regarding factors.

Feel free to edit as u like!

Best, Jakob

### CODE REVIEW

File: forcats.Rmd [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+- [Homepage](https://forcats.tidyverse.org/)
+- [Cheatsheet](https://github.com/rstudio/cheatsheets/raw/master/factors.pdf)
+- [Chapter in R4DS](https://r4ds.had.co.nz/factors.html)
+- [Vignette](https://forcats.tidyverse.org/articles/forcats.html)
+
+In this chapter the demo data set "diamonds" from the ggplot2-package ([more information](https://ggplot2.tidyverse.org/reference/diamonds.html)) and a dataset "olympic-games" which shows the medal success of olympic athletes from 1896 to 2016 are used. Before you start reading, you should have read the chapter ***Wrangling Data: dplyr*** or already have basic knowledge in this field. 
+
+```{r message=FALSE, warning=FALSE, paged.print=TRUE}
+library(tidyverse)
+diamonds<-read_csv("data/diamonds.csv")
```
Comment (carlschmidt26) [inline forcats.Rmd]: I would suggest to use spaces to highlight important operators.
E.g. I would change ```diamonds<-read_csv("data/diamonds.csv")``` to  ```diamonds <- read_csv("data/diamonds.csv")```.

The latter tends to be more readable. See also the [Style Guide](http://adv-r.had.co.nz/Style.html) of Advanced R by Hadley Wickham. 

In my opinion the same goes for the pipe operator `%>%` which isn't explicitly mentioned in the guide. 
The pipe operator will also be automatically inserted with blank spaces using the `Ctrl` + `Shift` + `M` shortcut.

File: forcats.Rmd [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  geom_col()
+```
+
+A sorting of the x-axis, which is based on the degree of clarity, is unfortunately not possible in this form. A workaround is to convert this column into a factor, which allows us to evaluate the individual categories. For this purpose we first define a vector, which ranks the categories according to their value:
+
+```{r}
+levels_clarity<-c("I1", "SI2", "SI1", "VS2", "VS1", "VVS2", "VVS1", "IF")
+```
+
+In the next step we lay this newly created levels vector on the column "clarity" of our data set to create a factor. This is a base-R function.
```
Comment (carlschmidt26) [inline forcats.Rmd]: Make clear which function is a base-R function.

File: forcats.Rmd [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+```{r}
+factor_clarity<-factor(x = diamonds$clarity, levels = levels_clarity)
+```
+
+Another possibility to create a factor is to convert a vector using ***as_factor().***. Here the levels are created automatically. Their order depends on the appearance of the corresponding value in the source vector.
```
Comment (carlschmidt26) [inline forcats.Rmd:47]: Maybe add an example of the order in `diamonds$clarity` before applying `as_factor()`

**Review by carlschmidt26 — COMMENTED** [review]
Hi Jakob,

I added some remarks. They are just for you to be aware of these points.
I am happy to work on theses issues if we agree to apply any changes.

Best Carl

File: forcats.Rmd (hunk 1 above)
Comment (jakobliebmann) [reply forcats.Rmd]: That's fine...i'll put spaces in the code

File: forcats.Rmd (hunk 2 above)
Comment (jakobliebmann) [reply forcats.Rmd]: Right, that's misleading. I'll solve this

File: forcats.Rmd (hunk 3 above)
Comment (jakobliebmann) [reply forcats.Rmd:47]: thought I had done that with my first plot....

**Comment by jakobliebmann** [comment]
Thx for your comments Carl! I'll go on it. One thing to think about: is it realy necessary to outline the original order once more?

**Comment by carlschmidt26** [comment]
> Thx for your comments Carl! I'll go on it. One thing to think about: is it realy necessary to outline the original order once more?

The output of the code chunks isn't printed. If the original order is already somewhere close by, it should be fine. Let's come back to that when I've found the time to have a look at what R Studio produces.

**Comment by jakobliebmann** [comment]
5 minutes ago a merge conflict appeared as till created the file "forcats.Rmd"...i solved this, merging should now be possible after review

**Review by carlschmidt26 — APPROVED** [review]
Document looks nice and neat.

**Pull request merged**